internal/product/handler.go

```
@@ -115,26 +116,29 @@ func (h *productHandler) GetProductsByPlatform(w http.ResponseWriter, r *http.Re
 
-	w.Header().Set("Content-Type", "application/json")
-	if err := json.NewEncoder(w).Encode(products); err != nil {
-		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(products); err != nil {
+		internal.HandleHttpError(w, internal.ErrorEnvelope{Detail: "Failed to encode response"}, http.StatusInternalServerError)
 		return
 	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(buf.Bytes())
```

golangci-lint (2.12.2)

[error] Error return value of `w.Write` is not checked

(errcheck)